spec/system/admin/products_v3/products_spec.rb

```diff
@@ -1032,4 +1143,41 @@ def expect_page_to_have_image(url)
   def tax_category_column
     @tax_category_column ||= '[data-controller="variant"] > td:nth-child(10)'
   end
+
+  def validate_tomselect_without_search!(page, field_name, value, search_selector)
+    tomselect_wrapper = page.find_field(field_name).sibling(".ts-wrapper")
+    tomselect_wrapper.find(".ts-control").click
+
+    expect(page).not_to have_selector(search_selector)
+
+    tomselect_wrapper.find(:css, '.ts-dropdown .ts-dropdown-content .option',
+                           text: value).click
+  end
```

Comment:
I think this is doing too much, it's checking if search is set as well selecting the value. It would be best if that was split in two. We have helper to interact with tomselect : 
https://github.com/openfoodfoundation/openfoodnetwork/blob/442991ac7b60b191bd58e5f57d8013f97ab00a79/spec/support/request/web_helper.rb#L100-L113

Reply:
Yup, @rioug, I took the idea from these methods. I just added the `expect` statement just after opening the dropdown in these methods implementation.😅
The scenarios I wanted to test are these:
- The search field **should not** be visible when there are **under** 11 items there
- The search field **should** be visible when there are **over** 11 items

Please let me know if I'm understanding it wrong:
- By dividing into two, do you mean?:

```ruby
  def open_tomselect_dropdown(page, field_name)
    tomselect_wrapper = page.find_field(field_name).sibling(".ts-wrapper")
    tomselect_wrapper.find(".ts-control").click

    tomselect_wrapper
  end

  def select_option(tomselect_wrapper)
    tomselect_wrapper.find(:css, '.ts-dropdown .ts-dropdown-content .option',
                           text: value).click
  end

  def validate_tomselect_without_search!(page, field_name, value, search_selector)
    tomselect_wrapper = open_tomselect_dropdown(page, field_name)

    expect(page).not_to have_selector(search_selector)

    select_option(tomselect_wrapper)
  end
```

Thanks.

Reply:
What I mean is it's not obvious that `validate_tomselect_without_search!` is also selecting the option, it would be better to have something like this : 
```ruby
def validate_tomselect_without_search(page, field_name, search_selector)
    tomselect_wrapper = page.find_field(field_name).sibling(".ts-wrapper")
    tomselect_wrapper.find(".ts-control").click

    expect(page).not_to have_selector(search_selector)
    # We might need to close the tomselect here, I am not sure.
end

# in the spec example you would do something like :
validate_tomselect_without_search(page, "Producer", producer_search_selector)

tomselect_select(producer_to_select, options)

```
This way it's obvious where the selection happens. Does that make sense ?

Reply:
`±` Good idea. 👍

Reply:
>  Does that make sense ?

Sure, Gaetan. That's a better separation of code indeed, however, I had been concerned with this approach before:
- `validate_tomselect_without_search` will have to close the Tomselect
- Because `tomselect_select` clicks on field to open the dropdown
- Due to the above steps, the spec would be like this:
  - First, the `validate_tomselect_without_search` method will open just to validate the presence of the search field
  - After that, the same process is repeated by `tomselect_select` to select the option
- That's why I came up with new methods to with a thought; why not put an `expect` in between 😂😂

However, no worries, I now see that I was overthinking 😅. We have much faster machines so it wouldn't be a problem. 
Let me incorporate the suggestion and get back to you, Thanks 🙌🏻

Reply:
>  Does that make sense ?

Sure, @rioug. That's a better separation of code indeed, however, I had been concerned with this approach before:
- `validate_tomselect_without_search` will have to close the Tomselect
- Because `tomselect_select` clicks on field to open the dropdown
- Due to the above steps, the spec would be like this:
  - First, the `validate_tomselect_without_search` method will open just to validate the presence of the search field
  - After that, the same process is repeated by `tomselect_select` to select the option
- That's why I came up with new methods to with a thought; why not put an `expect` in between 😂😂

However, no worries, I now see that I was overthinking 😅. We have much faster machines so it wouldn't be a problem. 
Let me incorporate the suggestion and get back to you, Thanks 🙌🏻

Reply:
>  Does that make sense ?

Sure, @rioug. That's a better separation of concerns indeed, however, I had been concerned with this approach before:
- `validate_tomselect_without_search` will have to close the Tomselect
- Because `tomselect_select` clicks on the field to open the dropdown
- Due to the above steps, the spec would be like this:
  - First, the `validate_tomselect_without_search` method will open just to validate the presence of the search field
  - After that, the same process is repeated by `tomselect_select` to select the option
- That's why I came up with new methods with a thought; why not put an `expect` in between 😂😂

However, no worries, I now see that I was overthinking 😅 and this way it's more readable and clear. We have much faster machines so it wouldn't be a problem.
Let me incorporate the suggestion and get back to you, Thanks 🙌🏻

Reply:
Thanks Ahmed, I think that makes a lot of sense, go for it!